crypto/objects/obj_dat.c

```diff
@@ -670,7 +670,7 @@ int OBJ_create_objects(BIO *in)
             }
         } else
             s = NULL;
-        if ((o == NULL) || (*o == '\0'))
+        if (s == NULL || *o == '\0')
```

Comment:
Isn't check for s == NULL sufficient? At least if `*o == '\0'`, then s will be NULL...

Reply:
Actually *formally* speaking this changes behaviour and question is if original behaviour, i.e. call to OBJ_create with s being NULL is intentional. If it is, then the suggested change would actually be justified... Or in other words have you examined if OBJ_create with s being NULL is bound to fail?

Reply:
> If it is [intentional], then the suggested change would actually be justified...

I meant "unjustified".

Reply:
Well, given the fact that in case of s being NULL OBJ_create could be called with garbage l, it's probably appropriate to assume that at least *that* is a bug. Yet it doesn't actually lift all concerns about original behaviour. Original code was looping till empty string was returned by BIO_gets. While suggested modification will terminate the loop at first string that doesn't meet criteria...

Reply:
There is no error management in this corner of code, either from callers or into this called method ...
Should we silently ignore a invalid line in the middle of file ? or reject It ?
Suggestion : add code to count number of file line, and compare it  to `num` (the oid creation count er) before return exit . so it may return -1 on error ?

Reply:
> There is no error management in this corner of code

Which is kind of argument in favour of preserving the original behaviour at least in released branches, but probably even master [since it's meant to be minor release]. Please note that I don't have ready answer...